pyccel/parser/semantic.py

```diff
@@ -2348,12 +2331,18 @@ def _visit_FunctionCall(self, expr):
                 errors.report(UNDEFINED_INIT_METHOD, symbol=name,
                 bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                 severity='error')
-            args = expr.args
-
+            d_var = {'datatype': self.get_class_construct(method.cls_name)(),
+                    'memory_handling':'stack',
+                    'shape' : None,
+                    'rank' : 0,
+                    'is_target' : False,
+                    'cls_base' : self.scope.find(method.cls_name, 'classes')}
+            cls_variable = self._assign_lhs_variable(expr.current_user_node.lhs, d_var, expr, [], True)
+            args = (FunctionCallArgument(cls_variable), *expr.args)
```

Comment:
I've just noticed that we are using `expr.args` here instead of `args` (the semantic objects created here: https://github.com/pyccel/pyccel/blob/devel/pyccel/parser/semantic.py#L2318). Please fix this

Reply:
Fixed